src/content_sao.cpp

```diff
@@ -969,6 +969,29 @@ void PlayerSAO::step(float dtime, bool send_recommended)
 			setBreath(m_breath + 1);
 	}
 
+	if (m_node_hurt_interval.step(dtime, 1.0)) {
+		// Feet, middle and head
+		v3s16 p1 = floatToInt(m_base_position + v3f(0, BS*0.1, 0), BS);
+		MapNode n1 = m_env->getMap().getNodeNoEx(p1);
+		v3s16 p2 = floatToInt(m_base_position + v3f(0, BS*0.8, 0), BS);
+		MapNode n2 = m_env->getMap().getNodeNoEx(p2);
+		v3s16 p3 = floatToInt(m_base_position + v3f(0, BS*1.6, 0), BS);
+		MapNode n3 = m_env->getMap().getNodeNoEx(p3);
+
+		u32 damage_per_second = 0;
+		damage_per_second = MYMAX(damage_per_second,
+			((Server*) m_env->getGameDef())->ndef()->get(n1).damage_per_second);
```

Comment:
No, no, no.

Please think of encapsulation and separation of concerns here!

Why should a PlayerSAO ever need to know about the map, or an nodedef?  You're making a twisted web of object dependencies.  There should be a clear and distinct heirarchy.

If you find yourself needing to write code like "((Server*) m_env->getGameDef())->ndef()->get(n1)", you know you're doing it wrong.

This is not acceptable to me.